### PR TITLE
Angular 4 upgrade

### PR DESCRIPTION
**Pull request by Blackbaud-SteveBrush**

### CODE REVIEW

**Review by blackbaud-brandonhare — APPROVED** [review]
LGTM

File: e2e/home.component.e2e-spec.ts [hunk 1]
```diff
@@ -11,9 +11,11 @@ describe('Home Component', () => {
...
     // Host URL + SPA Name automatically included
     SkyHostBrowser.get('/');
-    expect(element(by.tagName('h1')).getText()).toBe('Stache 2');
 
-    // // Since we've said this is an async test, we need to call done
-    done();
+    element(by.tagName('h1')).getText()
+      .then((text: string) => {
+        expect(text).toBe('Stache 2');
+        done();
```
Comment (Blackbaud-BrandonJones) [inline e2e/home.component.e2e-spec.ts:18]: This is all well and good for when we create this template... but if the users change this text in their project (as they should) won't this cause issues and failures when they try to publish?

Comment (Blackbaud-SteveBrush) [reply e2e/home.component.e2e-spec.ts:18]: Yep, but the idea is that they need to update the tests to support their site, or remove them altogether. The vanilla template expects the same things from their users: https://github.com/blackbaud/skyux-template/blob/master/e2e/home.component.e2e-spec.ts#L15

Comment (blackbaud-brandonhare) [reply e2e/home.component.e2e-spec.ts:18]: True. I still think we should rip this test out completely, but if we're going to keep it, it looks ok.

**Pull request merged**